plugins/cilium-cni/cni-install.sh

```diff
@@ -72,8 +79,42 @@ fi
 
 cp "/opt/cni/bin/${BIN_NAME}" "${CNI_DIR}/bin/"
 
+# Remove any active Cilium CNI configurations left over from previous installs
+# to make sure the one we're installing later will take effect.
+# Ignore the file specified by CNI_CONF_NAME. The agent will use this
+# filename to write a user-specified CNI config and races against this script.
+echo "Removing active Cilium CNI configurations from ${CNI_CONF_DIR}})..."
+find "${CNI_CONF_DIR}" -maxdepth 1 -type f \
+  -name '*cilium*' -and \( \
+    -name '*.conf' -or \
+    -name '*.conflist' \
+  \) \
+  -not -name "${CNI_CONF_NAME}" \
+  -delete
+
+# Rename all remaining CNI configurations to *.cilium_bak. This ensures only
+# Cilium's CNI plugin will remain active. This makes sure Pods are not
+# scheduled by another CNI when the Cilium agent is down during upgrades
+# or restarts. See GH-14128 and related issues for more context.
+if [ "${CNI_EXCLUSIVE}" != "false" ]; then
```

Comment:
Was the intent here to force users to opt out of exclusive control over the CNI?

I noticed that if the user fails to update their YAML files during the upgrade to this version, `CNI_EXCLUSIVE` will be undefined, and therefore does not match `false` and automatically takes this path.

Reply:
> Was the intent here to force users to opt out of exclusive control over the CNI?

Correct, it's opt-out since it can cause pods to be scheduled by other CNIs during Cilium upgrade/downtime. This is a potential security issue since network policies won't be enforced with those other CNIs. This has been documented in the upgrade notes. We didn't feel like this kind of change was appropriate to backport given its magnitude.

Didn't know about the `upgrade-impact` label, thanks!